src/log.rs

```diff
@@ -28,15 +38,29 @@ pub fn show(mut connection: Connection, verbose: bool) {
                 } else {
                     Time::Local(Local::now().naive_local())
                 };
-                let qos = format::qos(publish.qos);
                 let topic = publish.topic;
                 let size = publish.payload.len();
                 let payload = Payload::unlimited(publish.payload.into());
-                println!("{time:12} QoS:{qos:11} {topic:50} Payload({size:>3}): {payload}");
+
+                if json {
+                    match serde_json::to_string(&JsonLog {
+                        time,
+                        qos: publish.qos as u8,
+                        topic,
+                        size,
+                        payload,
+                    }) {
+                        Ok(out) => println!("{out}"),
+                        Err(err) => eprintln!("Failed to format message: {err}"),
```

Comment:
Thinking about this error case… is it possible to have an error here? I think I would just go with `…::to_string(…).expect("Should be able to format log line as JSON")` here. If this fails there is something wrong with the code, not with the data? Or am I overlooking a possible valid error case?